### PR TITLE
102 Automated SSL/TLS tests

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?
- Makes E2E someketest run against TLS-protected Conjur instance in E2E smoketests
- Cleans up a few related files

### What ticket does this PR close?

Connected to #102 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: examples/puppetmaster/code/environments/production/manifests/site.pp [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+  $secret = conjur::secret('inventory/db-password')
 
-    notify { "Writing secret '${secret.unwrap}' to /tmp/test.pem...": }
-    file { '/tmp/test.pem':
-      ensure  => file,
-      content => conjur::secret('inventory/db-password'),
-    }
-
-    notify { "Done!": }
+  notify { "Writing secret '${secret.unwrap}' to $pem_file...": }
```
Comment (izgeri) [inline examples/puppetmaster/code/environments/production/manifests/site.pp:18]: should this have a comment that indicates this is a bad idea to ever do in a real environment?

Comment (izgeri) [reply examples/puppetmaster/code/environments/production/manifests/site.pp:18]: hint: yes

File: examples/puppetmaster/smoketest_e2e.sh [hunk 2]
```diff
@@ -62,6 +68,11 @@ wait_for_puppetmaster() {
   echo "OK"
 }
 
+install_required_module_dependency() {
+  echo "Installing puppetlabs-registry module dep to server..."
+  docker-compose exec -T puppet puppet module install puppetlabs-registry
```
Comment (izgeri) [inline examples/puppetmaster/smoketest_e2e.sh:73]: why is this required?

File: examples/puppetmaster/code/environments/production/manifests/site.pp (hunk 1 above)
Comment (sgnn7) [reply examples/puppetmaster/code/environments/production/manifests/site.pp:18]: will add

File: examples/puppetmaster/smoketest_e2e.sh (hunk 2 above)
Comment (sgnn7) [reply examples/puppetmaster/smoketest_e2e.sh:73]: some (eg. windows) use of our plugin fails otherwise. it's a dependency that isn't automatically resolved given the way we add our plugin directly.

File: examples/puppetmaster/code/environments/production/manifests/site.pp (hunk 1 above)
Comment (sgnn7) [reply examples/puppetmaster/code/environments/production/manifests/site.pp:20]: fixed

File: examples/puppetmaster/smoketest_e2e.sh (hunk 2 above)
Comment (izgeri) [reply examples/puppetmaster/smoketest_e2e.sh:73]: is this something that needs to be documented in the README (eg for windows users)? can this PR include more context for why this change was added?

Comment (sgnn7) [reply examples/puppetmaster/smoketest_e2e.sh:73]: When installing from puppet forge, this isn't needed as the dependency should be automatically resolved since we have it in the [explicit dependency list](https://github.com/cyberark/conjur-puppet/blob/master/metadata.json#L11).
